src/checkers/htmlchecker.py

```diff
@@ -109,6 +108,19 @@ def check(self, external_data):
 
         self._update_version(external_data, latest_version, abs_url)
 
+    def _substitute_placeholders(self, template, version):
+        version_list = LooseVersion(version).version
+        number_of_fields = len(version_list)
+        for i in range(number_of_fields):
+            template = template.replace("$version" + str(i), str(version_list[i]))
+        template = template.replace("$version", str(version))
+        template = template.replace("$major", str(version_list[0]))
+        if number_of_fields > 1:
+            template = template.replace("$minor", str(version_list[1]))
+        if number_of_fields > 2:
+            template = template.replace("$patch", str(version_list[2]))
+        return template
```

Comment:
Better construct some `replacements` dict and use `Template.substitute(**replacements)` once instead of using `str.replace` consequently. The later doesn't take syntax into account and can replace unwanted substrings.

Reply:
Should the dict contain the keys version, major, minor and patch and Template.substitute be called after the loop?

Reply:
Yes, something like that, e.g. `{"major": version_list[0], "minor": version_list[1]}` and so on.

Reply:
I think something like this should do (you'll need to add string.Template import back):
```suggestion
        tmpl = Template(template)
        tmpl_vars = {"version": version}
        for i, version_part in enumerate(version_list):
            tmpl_vars[f"version{i}"] = version_part
            if i == 0:
                tmpl_vars["major"] = version_part
            elif i == 1:
                tmpl_vars["minor"] = version_part
            elif i == 2:
                tmpl_vars["patch"] = version_part
        return tmpl.substitute(**tmpl_vars)
```